### PR TITLE
fix(security): stored XSS via product SKU & product IDOR via id param

### PR DESCRIPTION
**Pull request by MdAsifHossainNadim**
### All Submissions:

* [x] My code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [x] My code satisfies feature requirements
* [x] My code is tested
* [x] My code passes the PHPCS tests
* [x] My code has proper inline documentation
* [x] I've added proper labels to this pull request

### Changes proposed in this Pull Request:

Fixes two security vulnerabilities reported against Dokan Lite (≤ 5.0.4). Both are tracked in the Dokan Pro security tracker but the vulnerable code lives in Lite.

---

#### Issue 1 — Stored XSS via Product SKU
🔗 getdokan/dokan-pro#5758 (CVE-2026-11783)

**What's the actual issue?**
A vendor controls their product **name** and **SKU**. The store-search AJAX handler (`Product\Hooks::store_product_search_action()`) builds an HTML snippet that the browser injects via jQuery `.html()`, but those values were concatenated **without escaping**. A vendor could set `SKU = ">` and have the script execute for anyone who searches that store — including logged-out visitors.

**How we fixed it** — `includes/Product/Hooks.php`
Escaped every vendor-controlled value at output:
```php
'<a href="' . esc_url( get_post_permalink( $result->ID ) ) . '">'
'<h3>' . esc_html( $get_name ) . '</h3>'
'<span>' . esc_html( $parent->name ) . '</span>'
'<span>' . esc_html( $category->name ) . '</span>'
'…SKU:… ' . esc_html( $sku ) . '</div>'
```
(`wc_price()` output is intentional HTML and is left as-is.)

**How to test manually**
1. As a vendor, set a product SKU to `">`.
2. Search that product on the store page (or POST to `admin-ajax.php?action=dokan_store_product_search_action`).
3. Before: alert fires. After: the SKU renders as inert text (`&lt;img…`).

**Test result:** ✅ Payload returned escaped (`&lt;img`, `&lt;script&gt;`); price/category/permalink render normally; search still works.

---

#### Issue 2 — IDOR / information disclosure via `id`
🔗 getdokan/dokan-pro#5773 (CVE-2026-11987)

**What's the actual issue?**
`GET /wp-json/dokan/v1/products?id=<X>` passed the request `id` straight to `WP_Query` as `author`, and the single-item permission check only verified the generic vendor capability. A vendor could read other vendors' products (incl. drafts) by changing `id`.

**How we fixed it**
- `includes/Abstracts/DokanRESTController.php` — admin-gate the `id`→`author` mapping (non-admins are always scoped to their own objects):
```php
$is_store_admin          = current_user_can( dokana_admin_menu_capability() );
$can_target_other_vendor = $is_store_admin && isset( $request['id'] );
$args['author'] = $can_target_other_vendor ? (int) $request['id'] : dokan_get_current_user_id();
```
- `includes/REST/ProductController.php` — enforce per-product ownership on the single-item read:
```php
if ( current_user_can( 'manage_options' ) ) { return true; }
if ( ! current_user_can( 'dokandar' ) ) { return false; }
if ( ! dokan_is_product_author( (int) ( $request['id'] ?? 0 ) ) ) {
    return new WP_Error( 'dokan_rest_cannot_view', __( 'Sorry, you are not allowed to view this product.', 'dokan-lite' ), [ 'status' => rest_authorization_required_code() ] );
}
return true;
```

**How to test manually**
1. As Vendor A, `GET /wp-json/dokan/v1/products?id=<Vendor B>` → before: B's products incl. drafts; after: only A's.
2. `GET /wp-json/dokan/v1/products/<B's product>` → after: `403`. As admin, both still work.

**Test result:** ✅ Vendor scoped to self; admin override preserved; single-item `403` for non-owner. Adversarial review across ~20 base-class inheritors found **0 regressions**.

---

### Related Pull Request(s)

* Dokan Pro security fixes (SQLi + privilege escalation): https://github.com/getdokan/dokan-pro/pull/5786

### Closes

* Closes getdokan/dokan-pro#5758
* Closes getdokan/dokan-pro#5773

### Changelog entry

**Title:** Fix stored XSS via product SKU and IDOR on the products REST endpoint

Previously the store product-search AJAX response did not escape vendor-controlled values (stored XSS via SKU/name), and the `dokan/v1/products` endpoint honored a request-supplied `id` as the author without an ownership check (IDOR exposing other vendors' products, including drafts). This PR escapes all vendor-controlled output and scopes product queries/reads to the owner (store admins excepted).

### Verification

* PHPCS: 0 errors on all changed files.
* Runtime-verified: XSS payloads returned escaped; vendor product queries/reads scoped to the owner; admin override intact.
* Adversarial multi-agent review: 0 regressions, fixes confirmed complete.

## Summary by CodeRabbit

* **Bug Fixes**
  * Tightened vendor and product query filtering to prevent unauthorized access to other vendors’ products and data.
  * Updated single-product REST authorization to use per-product ownership checks.
  * Corrected admin capability gating for product access and reverse withdrawal transaction responses.
  * Hardened AJAX store product search output by escaping vendor-controlled values in rendered markup.
* **Deprecations**
  * Marked the legacy admin menu capability helper as deprecated and emits a deprecation warning, advising the newer capability helper.

### CODE REVIEW

File: includes/Abstracts/DokanRESTController.php [hunk 1]
```diff
@@ -293,10 +293,14 @@ protected function prepare_data_for_response( $object, $request ) {
      * @return array
      */
     protected function prepare_objects_query( $request ) {
+        // Vendors are always scoped to their own objects; only a store admin may target another vendor via the id param.
+        $is_store_admin          = current_user_can( dokana_admin_menu_capability() );
```
Comment (mrabbani) [inline includes/Abstracts/DokanRESTController.php]: `dokana_admin_menu_capability` is the typo error `dokana` in this function.
- Deprecate this function
- Use the alternative function `dokan_admin_menu_capability`

**Review by mrabbani — CHANGES_REQUESTED** [review]

**Review by mrabbani — APPROVED** [review]
